solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -246,23 +247,53 @@ protected NamedList loadConfigSetFlags(CoreDescriptor cd, SolrResourceLoader loa
   public abstract String configSetName(CoreDescriptor cd);
 
   /**
-   * Check whether a config exists
+   * Upload files from a given path to config
    *
-   * @param configName the config to check existance on
-   * @return whether the config exists or not
-   * @throws IOException if an I/O error occurs
+   * @param dir        {@link java.nio.file.Path} to the files
+   * @param configName the name to give the config
+   * @throws IOException if an I/O error occurs or the path does not exist
    */
-  public boolean checkConfigExists(String configName) throws IOException {
-    return listConfigs().contains(configName);
-  }
+  public abstract void uploadConfig(Path dir, String configName) throws IOException;
```

Comment:
Let's have a consistent/purposeful approach to parameter ordering in this class.  We could make "configName" first *always*.  Or, we could vary it to go last *only* when we uploading (e.g. imagine an arrow showing the direction of the data from first param to second).